src/wp-includes/comment.php

```diff
@@ -2447,14 +2447,43 @@ function wp_new_comment_notify_postauthor( $comment_id ) {
 		return false;
 	}
 
-	// Only send notifications for approved comments.
-	if ( ! isset( $comment->comment_approved ) || '1' !== $comment->comment_approved ) {
-		return false;
+	// Send notifications for approved comments and all notes.
+	if (
+		! isset( $comment->comment_approved ) ||
+		( '1' !== $comment->comment_approved && 'note' !== $comment->comment_type ) ) {
+			return false;
 	}
 
 	return wp_notify_postauthor( $comment_id );
 }
 
+/**
+ * Send a notification to the post author when a new note is added via the REST API.
+ *
+ * @since 6.9.0
+ *
+ * @param WP_Comment $comment The comment object.
+ */
+function wp_new_comment_via_rest_notify_postauthor( $comment ) {
+	if ( 'note' === $comment->comment_type ) {
+		// Queue up the email notification as a cron callback. The second parameter ensures each callback is executed.
+		wp_schedule_single_event( time(), 'wp_new_comment_via_rest_notify_postauthor', array( $comment->comment_ID, time() ) );
+	}
+}
+
+/**
+ * Cron callback to send an author email notifcation, keeping the action asynchronous.
+ */
```

Comment:
```suggestion
/**
 * Cron callback to send an author email notifcation, keeping the action asynchronous.
 *
 * @since 6.9.0
 *
 * @param int $comment_id Comment ID.
 */
```

Reply:
Thanks for the feedback @mukeshpanchal27 - I wound up reverting this bit for now to keep the PR as simple as possible and the approach similar to comment notifications.